internal/database/repository/queue.go

```diff
@@ -82,7 +88,18 @@ func (s *PostgresRepository) UpsertQueues(ctx context.Context, queues []*dao.Par
 				"created_at":               time.Now().Unix(),
 			})
 		if err != nil {
-			return fmt.Errorf("could not insert/update queue into DB: %v", err)
+			return fmt.Errorf("could not insert/update queue %s into DB: %v", q.QueueName, err)
+		}
+
+		if len(q.Children) > 0 {
+			children := []*dao.PartitionQueueDAOInfo{}
+			for _, child := range q.Children {
+				children = append(children, &child)
+			}
+			err = s.UpsertQueues(ctx, parentId, children)
+			if err != nil {
+				return fmt.Errorf("could not one or more children of queue %s into DB: %v", q.QueueName, err)
```

Comment:
```suggestion
				return fmt.Errorf("could not upsert/insert one or more children of queue %s into DB: %v", q.QueueName, err)
```

Reply:
upsert is a combination of insert and update

Reply:
Fixed in commit 4473aa6